js/DataService.js

```diff
@@ -8,13 +8,46 @@ angular.module('PathOfDamage')
     var int = parseInt(string);
     return isNaN(int) ? null : int;
   }
+  
+  function Row(key, value) {
+    this.enabled = true;
+    this.name = '';
+    this.value = null;
+    if (key !== undefined) {
+      this[key] = value;
+    }
+    this.isEnabled = function () {
+      return this.enabled && this.value;
+    }
+  }
 
-  function checkTotal(table) {
-    if (table.totalMin) {
-      table.total = Math.max(table.total, table.totalMin);
+  function Table(name, addToTotal, key, value) {
+    this.name = name;
+    this.rows = [];
+    this.calcTotals = function () {
+      this.totals = this.getEmptyTotal();
+      for (var i = 0, len = this.rows.length - 1; i < len; i++) {
+        if (this.rows[i].isEnabled()) {
+          addToTotal(this.totals, this.rows[i]);
+        }
+      }
+    };
+    this.addRow = function () {
+      this.rows.push(new Row(key, value));
+    };
+    this.getEmptyTotal = function () {
+      return {total: 0};
     }
-    if (table.totalMax) {
-      table.total = Math.min(table.total, table.totalMax);
+  }
+
+  function TypedTable(name, addToSubtotal) {
+    Table.call(this, name, function (totals, row) {
+      for (var i = 0, len = row.types.length; i < len; i++) {
+        addToSubtotal(totals, row.value, row.types[i]);
+      }
+    }, 'types', ['physical']);
+    this.getEmptyTotal = function () {
+      return {};
```

Comment:
This causes the totals to end up in the order they're added in instead of a static order, but I wont' bother fixing it until after I look at the iconify stuff, where this logic may change.